drivers/sensor/bosch/bme280/bme280.c

```diff
@@ -321,14 +321,16 @@ static int bme280_chip_init(const struct device *dev)
 		return err;
 	}
 
-	err = bme280_reg_read(dev, BME280_REG_ID, &chip_id, 1);
-	if (err < 0) {
-		LOG_DBG("ID read failed: %d", err);
-		return err;
+	/* Give the ID read a few chances to succeed */
+	for (int i = 0; i < 5; i++) {
```

Comment:
> The trigger for this was the driver falling over because the first I2C transaction after boot happened to not read the ID correctly.

![image](https://github.com/user-attachments/assets/6aa2077b-4db9-4b99-8b36-769d50902585)


I would suggest instead adding the startup time the sensor spec indicates before performing any I2C-xfer. I don't think I've seen this retry-pattern used on any other sensor driver.

Reply:
The problem could not be due to this as the sensor is permanently powered.

Reply:
Could you share repro-steps?

Even if it's not the issue you're facing: not following the spec'd startup-time looks like a problem waiting to happen.

Reply:
I removed the change as it is not the core purpose of this PR. It may have been related to thread unsafety in the I2C drivers (#83020)